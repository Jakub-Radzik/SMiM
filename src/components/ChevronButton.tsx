import {StyleSheet, Text, TouchableOpacity, View} from 'react-native';

interface Props {
  title: string;
  onPress: () => void;
}

export const ChevronButton = ({title, onPress}: Props) => {
  return (
    <TouchableOpacity onPress={onPress}>
      <View style={styles.button}>
        <Text style={styles.chevron}>{title}</Text>
        <Text style={styles.chevron}>{'>'}</Text>
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  button: {
    padding: 10,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    borderTopWidth: 1,
    borderTopColor: 'lightgray',
  },
  chevron: {
    fontSize: 14,
  },
});
