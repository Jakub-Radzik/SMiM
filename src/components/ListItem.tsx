import {StyleSheet, Text, TouchableOpacity, View} from 'react-native';

interface ListItemProps {
  title: string;
  icon: string;
  onPress: () => void;
}

export const ListItem = ({title, icon, onPress}: ListItemProps) => {
  return (
    <TouchableOpacity onPress={onPress}>
      <View style={styles.container}>
        <Text style={styles.icon}>{icon}</Text>
        <Text style={styles.text}>{title}</Text>
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'space-between',
    alignItems: 'center',
    flexDirection: 'row',
    padding: 20,
    marginTop: 10,
    borderRadius: 10,
    backgroundColor: 'white',
    shadowOffset: {width: 0, height: 2},
    shadowColor: 'black',
    shadowOpacity: 0.26,
  },
  text: {fontSize: 20, letterSpacing: 2},
  icon: {fontSize: 30},
});
