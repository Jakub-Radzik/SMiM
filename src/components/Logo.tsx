import {Image} from 'react-native';
import logo from '../img/logo_t.png';

export const Logo = ({withHeader = true}: {withHeader?: boolean}) => (
  <Image
    source={logo}
    style={{
      position: 'absolute',
      top: withHeader ? 40 : 100,
      left: 20,
      opacity: 0.2,
      zIndex: -1,
    }}
  />
);
