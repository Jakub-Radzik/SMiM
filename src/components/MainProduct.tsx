import {View, StyleSheet, TouchableOpacity, Animated} from 'react-native';
import FastImage, {Source} from 'react-native-fast-image';
import LinearGradient from 'react-native-linear-gradient';
import {Product} from '../types';
import {Title} from './Title';
import {useSafeAreaInsets} from 'react-native-safe-area-context';
import {useEffect, useRef} from 'react';
import {Text} from './Text';

interface Props {
  product: Product;
  onPress: () => void;
}

export const MainProduct = ({product, onPress}: Props) => {
  const fadeAnim = useRef(new Animated.Value(0)).current; // Initial value for opacity: 0
  const pulseAnim = useRef(new Animated.Value(1)).current; // Initial value for opacity: 0

  useEffect(() => {
    Animated.timing(fadeAnim, {
      toValue: 1,
      duration: 1000,
      useNativeDriver: true,
    }).start();
  }, [fadeAnim]);

  useEffect(() => {
    Animated.loop(
      Animated.sequence([
        Animated.timing(pulseAnim, {
          toValue: 1.05,
          duration: 500,
          useNativeDriver: true,
        }),
        Animated.timing(pulseAnim, {
          toValue: 1,
          duration: 500,
          useNativeDriver: true,
        }),
      ]),
    ).start();
  }, [pulseAnim]);

  const {top} = useSafeAreaInsets();
  return (
    <TouchableOpacity style={styles.card} onPress={onPress}>
      <LinearGradient
        style={[styles.linearGradient, {paddingTop: top / 2}]}
        locations={[0, 0.7, 1]}
        colors={['#7a769f', '#4c669f', 'transparent']}>
        <Animated.View
          style={{
            width: '100%',
            opacity: fadeAnim,
            marginLeft: -90,
          }}>
          <FastImage
            source={product.images.thumbnail as unknown as Source}
            style={styles.image}
          />
        </Animated.View>
        <View style={styles.productDescription}>
          <Animated.Text style={{transform: [{scale: pulseAnim}]}}>
            <Title
              title="Hit tygodnia!"
              style={{color: 'white', fontSize: 40}}
            />
          </Animated.Text>
          <Text style={styles.product}>{product.title}</Text>
          <Text style={styles.price}>{product.price} $</Text>
        </View>
      </LinearGradient>
    </TouchableOpacity>
  );
};
const styles = StyleSheet.create({
  card: {
    marginBottom: 20,
  },
  image: {
    width: '100%',
    height: 300,
    zIndex: 2,
    marginTop: 20,
  },
  linearGradient: {
    paddingHorizontal: 15,
    borderRadius: 5,
    flexDirection: 'row',
    justifyContent: 'flex-start',
    alignItems: 'center',
    gap: 20,
  },
  productDescription: {
    gap: 10,
    alignItems: 'flex-end',
    marginTop: -100,
    marginLeft: -170,
  },
  product: {
    color: 'white',
    fontWeight: '500',
    fontSize: 20,
  },
  price: {
    color: '#F4B168',
    fontWeight: 'bold',
    fontSize: 30,
    marginTop: 30,
  },
});
