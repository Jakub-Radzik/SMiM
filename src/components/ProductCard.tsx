import {StyleSheet, Text, TouchableOpacity, View} from 'react-native';
import {Product} from '../types';
import FastImage from 'react-native-fast-image';

interface Props {
  item: Product;
  onPress: () => void;
}

export const ProductCard = ({item, onPress}: Props) => {
  return (
    <TouchableOpacity onPress={onPress}>
      <View style={styles.item}>
        <View style={styles.imageWrapper}>
          <FastImage source={item.images.thumbnail} style={styles.image} />
        </View>
        <Text style={styles.title}>{item.title}</Text>
        <Text style={styles.price}>{item.price} $</Text>
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  item: {
    padding: 10,
    alignItems: 'center',
    gap: 5,
  },
  imageWrapper: {
    backgroundColor: 'white',
    borderRadius: 10,
    paddingHorizontal: 15,
    paddingVertical: 2,
  },
  image: {
    width: 100,
    height: 100,
    resizeMode: 'contain',
  },
  title: {
    fontSize: 16,
    fontWeight: '500',
  },
  price: {
    fontSize: 14,
  },
});
