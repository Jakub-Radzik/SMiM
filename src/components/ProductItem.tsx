import {Image, StyleSheet, Text, TouchableOpacity} from 'react-native';
import {Product} from '../types';

type Props = {item: Product; onPress: () => void};

export const ProductItem = ({item, onPress}: Props) => {
  return (
    <TouchableOpacity onPress={onPress} style={styles.item}>
      <Image
        source={item.images?.thumbnail}
        style={{width: '100%', height: 150}}
      />
      <Text>{item.title}</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  item: {
    width: '100%',
    minHeight: 200,
    padding: 5,
    alignItems: 'center',
    gap: 10,
  },
});
