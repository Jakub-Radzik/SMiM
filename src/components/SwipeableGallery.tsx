import {useEffect, useState} from 'react';
import {
  Dimensions,
  ImageSourcePropType,
  StyleSheet,
  TouchableOpacity,
  View,
} from 'react-native';
import FastImage from 'react-native-fast-image';

interface Props {
  images: ImageSourcePropType[];
  link?: string;
  startNumber?: number;
  endNumber?: number;
}

const INTERVAL = 100; // Minimum swipe distance for number change
const IMAGE_HEIGHT = 300;
const IMAGE_WIDTH = Dimensions.get('window').width;

export const SwipeableGallery = ({images}: Props) => {
  const [currentIndex, setCurrentIndex] = useState(0);
  const [intervalId, setIntervalId] = useState<NodeJS.Timeout | null>(null);

  const clean = () => {
    if (intervalId) {
      clearInterval(intervalId);
    }
  };

  useEffect(() => {
    return () => {
      clean();
    };
  }, [intervalId]);

  const startInterval = (direction: 'forward' | 'backward') => {
    clean();

    const newIntervalId = setInterval(() => {
      setCurrentIndex(prevIndex => {
        if (direction === 'forward') {
          return (prevIndex + 1) % images.length;
        } else {
          return prevIndex === 0 ? images.length - 1 : prevIndex - 1;
        }
      });
    }, INTERVAL);

    setIntervalId(newIntervalId);
  };

  const next = () => {
    setCurrentIndex(prevIndex =>
      prevIndex === 0 ? images.length - 1 : prevIndex - 1,
    );
  };

  const prev = () => {
    setCurrentIndex(prevIndex => (prevIndex + 1) % images.length);
  };

  const stopInterval = () => {
    clean();
    setIntervalId(null);
  };

  return (
    <View style={styles.container}>
      <FastImage source={images[currentIndex]} style={styles.image} />
      <View style={styles.buttons}>
        <TouchableOpacity
          style={styles.button}
          onPress={prev}
          onLongPress={() => startInterval('forward')}
          onPressOut={stopInterval}></TouchableOpacity>
        <TouchableOpacity
          style={styles.button}
          onPress={next}
          onLongPress={() => startInterval('backward')}
          onPressOut={stopInterval}></TouchableOpacity>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    position: 'relative',
  },
  image: {
    width: IMAGE_WIDTH,
    height: IMAGE_HEIGHT,
    resizeMode: 'contain',
  },
  loadingContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  buttons: {
    position: 'absolute',
    top: 0,
    left: 0,
    flexDirection: 'row',
  },
  button: {
    width: '50%',
    height: IMAGE_HEIGHT,
  },
});
