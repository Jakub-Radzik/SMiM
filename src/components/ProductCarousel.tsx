import {View, ScrollView, StyleSheet} from 'react-native';
import {Category, Product} from '../types';
import {ChevronButton} from './ChevronButton';
import {ProductCard} from './ProductCard';
import {Title} from './Title';

interface Props {
  title: string;
  products: Product[];
  category: Category;
  onProductPress: (product: Product) => void;
  onChevronPress: () => void;
}

export const ProductCarousel = ({
  title,
  products,
  category,
  onProductPress,
  onChevronPress,
}: Props) => {
  return (
    <View style={styles.row}>
      <View style={styles.titleWrapper}>
        <Title title={title} />
      </View>
      <ScrollView horizontal showsHorizontalScrollIndicator={false}>
        {products.map((item, index) => (
          <ProductCard
            key={index}
            item={item}
            onPress={() => onProductPress(item)}
          />
        ))}
      </ScrollView>
      <ChevronButton title={'Zobacz wszystkie'} onPress={onChevronPress} />
    </View>
  );
};

const styles = StyleSheet.create({
  row: {
    paddingHorizontal: 10,
  },
  titleWrapper: {
    paddingHorizontal: 5,
  },
  button: {
    padding: 10,
    backgroundColor: 'white',
    borderRadius: 10,
    alignItems: 'center',
    justifyContent: 'center',
  },
});
