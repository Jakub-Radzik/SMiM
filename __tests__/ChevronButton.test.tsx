import React from 'react';
import renderer from 'react-test-renderer';
import {it, expect} from '@jest/globals';
import {ChevronButton} from '../src/components/ChevronButton';

const TITLE = 'Title';

it('chevron snapshot test', () => {
  const render = renderer
    .create(<ChevronButton title={TITLE} onPress={() => {}} />)
    .toJSON();

  expect(render).toMatchSnapshot();
});
