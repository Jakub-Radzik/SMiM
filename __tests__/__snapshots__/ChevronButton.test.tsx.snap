// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chevron snapshot test 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "borderTopColor": "lightgray",
        "borderTopWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 10,
      }
    }
  >
    <Text
      style={
        {
          "fontFamily": "Trebuchet MS",
          "fontSize": 14,
        }
      }
    >
      Title
    </Text>
    <Text
      style={
        {
          "fontFamily": "Trebuchet MS",
          "fontSize": 14,
        }
      }
    >
      &gt;
    </Text>
  </View>
</View>
`;

exports[`renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "borderTopColor": "lightgray",
        "borderTopWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 10,
      }
    }
  >
    <Text
      style={
        {
          "fontFamily": "Trebuchet MS",
          "fontSize": 14,
        }
      }
    >
      Title
    </Text>
    <Text
      style={
        {
          "fontFamily": "Trebuchet MS",
          "fontSize": 14,
        }
      }
    >
      &gt;
    </Text>
  </View>
</View>
`;
